src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```diff
@@ -159,6 +159,12 @@ public ConnectionSettings Settings
         }
 
         // methods
+        public void AddAdditionalReadTimeout(TimeSpan timeout)
+        {
+            ThrowIfDisposed();
+            _stream.ReadTimeout += (int)timeout.TotalMilliseconds;
```

Comment:
see for details: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/137/files#r438898945